### ISSUE TITLE
Q: Why P5 / 32 colors palette instead of 16 or 256 colors palettes?

### ISSUE THREAD

**Issue by CHiPs44** [MichaelBell/pico-stick#3]
Hello,

I can't say I understand all concepts and tricks involved in making this work, but why isn't 16 or 256 colors palette mode implemented?

16 colors should make it possible to have bitmaps with 2 pixels per byte, and handling several palettes (for example one for frame buffer and one for sprites or more) should make it possible to display more colors easily.

256 colors palette could be sufficient for many applications instead of diving into RGB555 that needs 2 bytes per pixel.

I see you define PALETTE_SIZE as 32 and there are LUTs for TMDS depending on this, 32×32×2×4 is 8192, 256×256×2×4 is 524288 which may be the reason, but I saw at least one reference to 256 colors palette...

**Comment by MichaelBell** [comment]
P5 was chosen as a good compromise between the size of the TMDS LUT required for full resolution encode, and number of colours supported.  As depth is also supported, 6 of the 8 bits per byte are used in this mode (the palette colour is in bits 6-2 and depth is in bit 0).  This choice of bits is again chosen to optimize performance!

I did implement 256 colour palette mode for horizontally pixel doubled mode only.  This doesn't have depth support, and can't be extended to the full resolution modes as the TMDS encode LUTs for each channel would be 512kB, as you have correctly surmised!

That 256 colour palette mode is not currently exposed in pimoroni-pico - I used it for the DOOM port and haven't got around to adding it there.
